### ISSUE TITLE
Lint consistency fuzz testing?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2191]
One concern I have about the consistency cleanups in this release (#2039, #2046, #2190) is there's no mechanism to ensure future improvements/new linters continue to keep "up to code" & enforce the same consistency.

One approach that seems doable is to have a GHA that [fuzzes](https://en.wikipedia.org/wiki/Fuzzing) our test suite. For example, for #2190, that means:

 - Go through the suite and identify all `expect_lint()` calls using "function(...)" or `\(...)` in the `content=` argument (text matching should be enough, but we could also use `getParseData()` here...).
 - Randomly (or exhaustively?) switch all such usages to their opposite: `s/function/\/g` and `s/\/function/g`
 - Run the test suite
 - Any test failure means a linter is inconsistently handling `function(...)` and `\(...)`

Similar fuzzes could apply in the other cases:

 - Randomly swap `%>%` and `|>`
 - Randomly swap `$` and `@`

Some exceptions will need to be carved out, e.g. for cases where intentionally only one alternative is meant to be linted, and we might need some work to make the fuzzing robust to changes in an expected lint message. But overall this approach seems pretty doable to me.

If run with randomness, I'd also run it periodically, not on push/merge, since a failure might not be caught up-front & surfacing that failure on "someone else's" PR would be bad form.

**Comment by AshesITR** [comment]
Interesting idea. Things to consider:

* Lint metadata may also change, especially expected column ranges.
* `|>` and `%>%` can't be swapped in all contexts; they also have different placeholder semantics (`_` vs. `.`)

**Comment by MichaelChirico** [comment]
True. Let's see how relevant those turn out in practice. Another idea is to allow for a `# nofuzz` escape to skip certain tests if they're too onerous to accomodate.

**Comment by MichaelChirico** [comment]
we can also work on #2221 here by fuzzing the lint metadata & making sure tests fail.

**Comment by MichaelChirico** [comment]
Another idea for the fuzz/metatesting suite, based on #2402 -- inject comments in random places in the test code and ensure the linters keep working. Have to exclude linters where `<COMMENT>` is meaningful, of course.

**Comment by AshesITR** [comment]
Another fuzzer idea: swap `=`, `<-` and `->`, and maybe the double-assignment arrows.

**Comment by IndrajeetPatil** [comment]
Of interest: there is also a new package for mutation testing in R (https://jakubsobolewski.com/muttest/)

**Comment by Bisaloo** [comment]
Another potential fuzzer: modify string formats. Switching `'` to `"` and try raw strings.

Linters that correctly use `get_r_string()` should pass fine but maybe we are missing cases where it should be used.

**Comment by MichaelChirico** [comment]
oh that's a really good one
